docs/data/data-grid/state/RowSelectorPlaygroundNoSnap.tsx

```diff
@@ -0,0 +1,511 @@
+import * as React from 'react';
+
+import { debounce } from '@mui/material/utils';
+
+import Box from '@mui/material/Box';
+import FormLabel from '@mui/material/FormLabel';
+import FormControl from '@mui/material/FormControl';
+import FormGroup from '@mui/material/FormGroup';
+import FormControlLabel from '@mui/material/FormControlLabel';
+import Checkbox from '@mui/material/Checkbox';
+import List from '@mui/material/List';
+import ListItem from '@mui/material/ListItem';
+import ListItemText from '@mui/material/ListItemText';
+import Slider from '@mui/material/Slider';
+import Typography from '@mui/material/Typography';
+import Accordion from '@mui/material/Accordion';
+import AccordionSummary from '@mui/material/AccordionSummary';
+import AccordionDetails from '@mui/material/AccordionDetails';
+
+import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
+
+import {
+  DataGridPro,
+  GridColumns,
+  GridRowsProp,
+  gridRowIdsSelector,
+  useGridApiRef,
+  GridEvents,
+  GridFilterModel,
+  gridSortedRowIdsSelector,
+  gridFilteredSortedRowIdsSelector,
+  gridVisibleSortedRowIdsSelector,
+  gridVisibleSortedTopLevelRowEntriesSelector,
+  gridPaginatedVisibleSortedGridRowEntriesSelector,
+} from '@mui/x-data-grid-pro';
+
+import HighlightedCode from 'docs/src/modules/components/HighlightedCode';
+
+const getTreeDataPath = (row) => [row.bicycleType, row.color];
+
+const allColumns: GridColumns = [
+  {
+    field: 'bicycleType',
+    headerName: 'Bicycle Type',
+    type: 'singleSelect',
+    valueOptions: ['Road Bike', 'Mountain Bike', 'Touring Bike', 'Folding Bike'],
+    width: 150,
+  },
+  {
+    field: 'color',
+    headerName: 'Color',
+    type: 'singleSelect',
+    valueOptions: ['blue', 'red', 'white', 'black'],
+    width: 100,
+  },
+  {
+    field: 'price',
+    type: 'number',
+    width: 150,
+  },
+  {
+    field: 'id',
+    headerName: 'row id',
+    flex: 1,
+  },
+];
+
+const rows: GridRowsProp = [
+  {
+    id: 'Touring-blue-945',
```

Comment:
I would have used integer IDs because they fit completely in the column. It's easier to match what will be returned with that is rendered. 

<img width="300" src="https://user-images.githubusercontent.com/42154031/160624492-10b8307d-afa6-4138-9408-27dee3894ba0.png" alt="image" style="max-width: 100%;">